Source/Core/Core/DSP/DSPDisassembler.cpp

```diff
@@ -105,8 +105,9 @@ std::string DSPDisassembler::DisassembleParameters(const DSPOPCTemplate& opc, u1
         // LSL, LSR, ASL, ASR
         if (opc.params[j].mask == 0x003f)
         {
-          // 6-bit sign extension
-          buf += fmt::format("#{}", (val & 0x20) != 0 ? (val | 0xFFFFFFC0) : val);
+          // Left and right shifts function essentially as a single shift by a 7-bit signed value,
+          // but are split into two intructions for clarity.
+          buf += fmt::format("#{}", (val & 0x20) != 0 ? (64 - val) : val);
```

Comment:
Note that the check with `val & 0x20` is a bit suspicious too; based on the interpreter code, the actual field is 6 bits, with the 7th bit (not included in the mask) acting as a sign bit.  Since the accumulators are 48 bits, there could be a shift by over 32 bits which would incorrectly trigger the current code (though I'm not sure this actually happens).  Since this code is already a special case for LSL/LSR/ASL/ASR, the 7th bit could be checked here, but it would be somewhat jank.